src/app/models/job-task-slot-counts.ts

```diff
@@ -0,0 +1,25 @@
+import { Model, Prop, Record } from "@batch-flask/core";
+
+export interface JobTaskSlotCountsAttributes {
+    active?: number;
+    running?: number;
+    completed?: number;
+    succeeded?: number;
+    failed?: number;
+}
+
+@Model()
+export class JobTaskSlotCounts extends Record<JobTaskSlotCountsAttributes> {
+    @Prop() public active: number = 0;
+    @Prop() public running: number = 0;
+    @Prop() public completed: number = 0;
+    @Prop() public succeeded: number = 0;
+    @Prop() public failed: number = 0;
+
+    public total: number;
+
+    constructor(data: JobTaskSlotCountsAttributes = {}) {
+        super(data);
+        this.total = this.active + this.running + this.completed;
```

Comment:
Should this be either a read-only, calculated getter, or a getter with a private setter?

Reply:
In other words, should it change and should we allow the consumer to be the one that changes it?

Reply:
I'm not too sure. I modeled this after the `job-task-counts.js` model to help create this one. If I'm understanding this correctly, these need to be public setters due to the deserialization that happens from this mapping call in the `getTaskCounts` function in this service: https://github.com/Azure/BatchExplorer/blob/2219e31cb8e623d3ce61d8a9784de0e99c28090b/src/app/services/azure-batch/job/job.service.ts#L138